README.md

```diff
@@ -33,13 +27,69 @@ The API should `CREATE`, `READ`, `UPDATE` and `DELETE` from a list of todos in a
-| `PATCH`* | `/todos/:id/complete` | Update todo, set attribute complete to `true` | `http://localhost:8080/todos/todos/01507581-9d12-4c3a-bb60-19d539a11189/complete` | | 200 | |
-| `PATCH`* | `/todos/:id/undo` | Update todo, set attribute complete to `false` |`http://localhost:8080/todos/todos/01507581-9d12-4c3a-bb60-19d539a11189/undo` | | 200 | |
-| `DELETE`* | `todos/:id` | Deletes a todo by `id` |`http://localhost:8080/todos/01507581-9d12-4c3a-bb60-19d539a11189` | | 200 | |
+| `POST`* | `/todos` | Add a new todo to the todo list | `http://localhost:8080/todos` | `{'name' : 'Buy oatmilk x 2', 'due':'2021-11-20T18:25:43.511Z'}` | 201 (Created), 400 if incorrect data submitted| |
+| `PATCH`* | `/todos/:id` | Edit the `name` and/or `due` date attributes of a todo.  | `http://localhost:8080/todos/:9d127581-0150-4c3a-bb60-1118919d539a` | `{"name":"Buy 3 Cartons of Oat Milk","due":"2021-12-23T18:25:43.511Z"}` | 200 (OK) or 400 (Bad Request) if invalid id or request attributes |  | `http://localhost:8080/todos/01507581-9d12-4c3a-bb60-19d539a11189` | | 200 | |
+| `POST`* | `/todos/:id/complete` | Update todo, set attribute complete to `true` | `http://localhost:8080/todos/todos/01507581-9d12-4c3a-bb60-19d539a11189/complete` | | 200 (OK) or 400 (Bad Request) if invalid id | |
+| `POST`* | `/todos/:id/undo` | Update todo, set attribute complete to `false` |`http://localhost:8080/todos/todos/01507581-9d12-4c3a-bb60-19d539a11189/undo` | | 200 (OK) or 400 (Bad Request) if invalid id | |
+| `DELETE`* | `todos/:id` | Deletes a todo by `id` |`http://localhost:8080/todos/01507581-9d12-4c3a-bb60-19d539a11189` | | 200 (OK) or 400 (Bad Request) if invalid id | |
+
+*_`POST`, `PUT`, `PATCH` and `DELETE` methods may need an API testing tool such as Postman to test._
+
+## Expected structure of a Todo
+
+```json
+  {
+    "id":"01507581-9d12-4c3a-bb60-19d539a11189",
+    "name":"Learn to use Adobe Photoshop",
+    "created":"2021-11-20T18:25:43.511Z",
+    "due":"2021-12-23T23:05:03.352Z",
+    "completed":false
+  }
+```
+
+## Setup and Configuration
+
+1. Create a file called `.env` in your home directory (outside of the `src` folder). Add the following text:
+
+```
+BASE_JSON_PATH="\\models\\todos.json"
+```
+This will ensure the current starter code is using the `todos.json` in the `models` folder.
+
+In your code when saving the updated list of todos back to the `todos.json` file, use the following environment variable:
+
+```javascript
+process.env.BASE_JSON_PATH
+```
+e.g.
+```javascript
+fs.writeFile(__dirname + process.env.BASE_JSON_PATH,  content, err => {
+    if (err) {
+      console.error(err)
+      return
+    }
+}
+```
+
+2. Clone this repository to your local machine. Once cloned, open in Visual Code. Run the following commands to install any dependencies:
```

Comment:
Does this step need to come before step 1?

Reply:
Yes, good shout. I'll move it now